org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -1368,98 +1363,32 @@ class CGenerator extends GeneratorBase {
      * @param the content of the "services" section of the docker-compose.yml file.
      * @param the name of the federate to be added to "services".
      * @param the name of the federate's Dockerfile.
-     * @param the federationID.
      */
-    def appendFederateToDockerComposeServices(StringBuilder dockerComposeServices, String federateName, String dockerFileName, int dockerComposeFederationID) {
+    def appendFederateToDockerComposeServices(StringBuilder dockerComposeServices, String federateName, String context, String rtiName, String dockerFileName) {
         val tab = '    '
         dockerComposeServices.append('''«tab»«federateName»:«System.lineSeparator»''')
         dockerComposeServices.append('''«tab»«tab»build:«System.lineSeparator»''')
-        dockerComposeServices.append('''«tab»«tab»«tab»context: «federateName»«System.lineSeparator»''')
+        dockerComposeServices.append('''«tab»«tab»«tab»context: «context»«System.lineSeparator»''')
         dockerComposeServices.append('''«tab»«tab»«tab»dockerfile: «dockerFileName»«System.lineSeparator»''')
-        dockerComposeServices.append('''«tab»«tab»command: -i «dockerComposeFederationID»«System.lineSeparator»''')
-    }
-
-    /**
-     * Write a Dockerfile for the RTI at rtiDir.
-     * The file will go into src-gen/RTI/rti.Dockerfile.
-     * @param the directory where rti.Dockerfile will be written to.
-     * @param name of the Dockerfile for the RTI.
-     */
-    def writeRTIDockerFile(File rtiDir, String dockerFileName) {
-        val dockerFile = rtiDir + File.separator + dockerFileName
-        // If a dockerfile exists, remove it.
-        var file = new File(dockerFile)
-        if (file.exists) {
-            file.delete
-        }
-        if (this.mainDef === null) {
-            return
+        dockerComposeServices.append('''«tab»«tab»command: -i 1«System.lineSeparator»''')
+        if (isFederated) {
+            dockerComposeServices.append('''«tab»«tab»depends_on: [«rtiName»]«System.lineSeparator»''')
```

Comment:
Nice find! This is pretty cool.

Reply:
One thing to note though, "depends_on" only makes sure the RTI container is "started" before the federate containers, so there is still a chance that the federates would start later, but try to connect to the RTI before it is ready..

Reply:
I guess it's about time we fixed the underlying issue :-)